onnxruntime/core/framework/execution_frame.h

```diff
@@ -199,8 +199,7 @@ class ExecutionFrame final : public IExecutionFrame {
                                                   bool create_fence);
 
   Status AllocateTensorWithPreAllocateBufferHelper(OrtValue& ort_value, void* pBuffer, MLDataType element_type,
```

Comment:
you don't need stride for training build?

Reply:
The call passes in its own shape, which may be different from original shape, so using the original strides might be long. Ideally we need to add strides to Output() call and all following functions to take new strides. For now, I am thinking we require user to set the strides accordingly after the Output() call, or do you think it's still better we add the parameter for all the functions? Note that we need the #ENABLE_TRAINING, which will make the code complex.